app/services/credentials_verifier.go

```diff
@@ -45,5 +47,16 @@ func CredentialsVerifier(store data.AccountStore, cfg *app.Config, username stri
 		return nil, FieldErrors{{"credentials", ErrExpired}}
 	}
 
+	//Check TOTP MFA
+	if account.TOTPEnabled() {
+		secret, err := compat.Decrypt([]byte(account.TOTPSecret.String), cfg.DBEncryptionKey)
```

Comment:
Check if TOTP is missing here and return special error if so similar to `ErrInvalidOrExpired`.